src/Controller/BackendCsvImportController.php

```diff
@@ -0,0 +1,356 @@
+     */
+    private function getFormId(Request $request)
+    {
+        return 'tl_csv_import_'.$request->query->get('key');
+    }
+
+    /**
+     * Returns the back button and redirect URL.
+     *
+     * @param Request $request
+     *
+     * @return string
+     */
+    private function getBackUrl(Request $request)
+    {
+        return str_replace('&key='.$request->query->get('key'), '', $request->getRequestUri());
+    }
+
+    /**
+     * Returns an array of separators for the template.
+     *
+     * @param bool $allowLinebreak
+     *
+     * @return array
+     */
+    private function getSeparators($allowLinebreak = false)
+    {
+        $separators = [
+            self::SEPARATOR_COMMA     => [
+                'delimiter' => ',',
+                'value'     => self::SEPARATOR_COMMA,
+                'label'     => $GLOBALS['TL_LANG']['MSC']['comma'],
+            ],
+            self::SEPARATOR_SEMICOLON => [
+                'delimiter' => ';',
+                'value'     => self::SEPARATOR_SEMICOLON,
+                'label'     => $GLOBALS['TL_LANG']['MSC']['semicolon'],
+            ],
+            self::SEPARATOR_TABULATOR => [
+                'delimiter' => "\t",
+                'value'     => self::SEPARATOR_TABULATOR,
+                'label'     => $GLOBALS['TL_LANG']['MSC']['tabulator'],
+            ],
+        ];
+
+        if ($allowLinebreak) {
+            $separators[self::SEPARATOR_LINEBREAK] = [
+                'delimiter' => "\n",
+                'value'     => self::SEPARATOR_LINEBREAK,
+                'label'     => $GLOBALS['TL_LANG']['MSC']['linebreak'],
+            ];
+        }
+
+        return $separators;
+    }
+
+    /**
+     * Converts a separator name/constant into a delimiter character.
+     *
+     * @param string $separator
+     *
+     * @return string
+     *
+     * @throws \RuntimeException
+     */
+    private function getDelimiter($separator)
+    {
+        $separators = $this->getSeparators(true);
+
+        if (!isset($separators[$separator])) {
+            throw new \RuntimeException($GLOBALS['TL_LANG']['MSC']['separator'][1]);
+        }
+
+        return $separators[$separator]['delimiter'];
+    }
+
+    /**
+     * Get uploaded files from FileUpload instance.
+     *
+     * @param FileUpload $uploader
+     *
+     * @return array
+     *
+     * @throws \RuntimeException
+     */
+    private function getFiles(FileUpload $uploader)
+    {
+        $files = $uploader->uploadTo('system/tmp');
+
+        if (count($files) < 1) {
+            throw new \RuntimeException($GLOBALS['TL_LANG']['ERR']['all_fields']);
+        }
+
+        foreach ($files as &$file) {
+            $extension = pathinfo($file, PATHINFO_EXTENSION);
+
+            if ('csv' !== $extension) {
+                throw new \RuntimeException(sprintf($GLOBALS['TL_LANG']['ERR']['filetype'], $extension));
+            }
+
+            // Drop TL_ROOT when we got rid of FileUpload which uses it
```

Comment:
The comment can be removed. It is missing the `TODO: ` prefix and I guess we are all aware of the `TL_ROOT` deprecation.